eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentLifecycle.scala

```diff
@@ -139,8 +139,17 @@ class PaymentLifecycle(nodeParams: NodeParams, progressHandler: PaymentProgressH
             }
             // in any case, we forward the update to the router
             router ! failureMessage.update
+            // we also update assisted routes, because they take precedence over the router's routing table
+            val assistedRoutes1 = c.assistedRoutes.map(_.map {
+              case extraHop: ExtraHop if extraHop.shortChannelId == failureMessage.update.shortChannelId => extraHop.copy(
+                cltvExpiryDelta = failureMessage.update.cltvExpiryDelta,
```

Comment:
I think you should copy almost everything (htlcMin and htlcMax are theoretically useful to copy because they play a small role in path-finding scoring iirc)

Reply:
I can't? This is an `ExtraHop`, which only has a subset of what `ChannelUpdate`'s fields.

Reply:
Indeed, got it.